src/commands/user-bonuses/list.ts

```diff
@@ -0,0 +1,95 @@
+import * as cosmos from "@azure/cosmos";
+import { Command } from "@oclif/command";
+import cli from "cli-ux";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+
+import { getCosmosConnection, pickAzureConfig } from "../../utils/azure";
+
+export default class UserBonusesList extends Command {
+  public static description = "Lists all user bonuses";
+
+  // tslint:disable-next-line:readonly-array
+  public static args = [
+    {
+      name: "fiscalCode",
+      required: false
+    }
+  ];
+
+  public static flags = {
+    ...cli.table.flags()
+  };
+
+  public async run(): Promise<void> {
+    const { args, flags: parsedFlags } = this.parse(UserBonusesList);
+
+    let fiscalCode: string = "";
```

Comment:
To avoid using let you could refactor as follows

```typescript
const fiscalCodeOrErrors = FiscalCode.decode(args.fiscalCode);
if (fiscalCodeOrErrors.isLeft()) {
  this.error("Invalid fiscalCode");
  return;
}
const fiscalCode = fiscalCodeOrErrors.value;
```

Reply:
args.fiscalCode is not required and fiscalCodeOrErrors.isLeft() returns error when args.fiscalCode is not specified.

Reply:
An alternative is to split into two function:
- user-bonuses:list (without FISCALCODE)
- user-bonuses:check FISCALCODE (FISCALCODE required)

Reply:
Could it be a solution?
```typescript
const fiscalCodeOrErrors = FiscalCode.decode(args.fiscalCode);
    // if fiscalCode is defined, check if it is in a valid format
    if (
      args.fiscalCode &&
      args.fiscalCode.length > 0 &&
      fiscalCodeOrErrors.isLeft()
    ) {
      this.error("Invalid fiscalCode");
      return;
    }
    const fiscalCode = fiscalCodeOrErrors.fold(
      () => "",
      fc => fc as string
    );
```

Reply:
yes thank you, it works.

I don't understand this but I'll put it in my todo study backlog :D

```ts
    const fiscalCode = fiscalCodeOrErrors.fold(
      () => "",
      fc => fc as string
    );
```

Reply:
That code means:
if `fiscalCodeOrErrors` is left (decoding fails cause the given fiscal code is not respecting that format), execute the first function `() => ""` (return an empty string)
otherwise if it is right (so the decoding succeeded) execute the second function that return the decoded value casted to a string `fc => fc as string`

So after that fiscalCode could have 2 value: empty string or the fiscal code in valid format

Reply:
many thanks @Undermaken